4.x: Generated validation type emits unreachable switch for OpenAPI interface

Generated validation type emits unreachable switch for OpenAPI interface

## Problem

A generated OpenAPI declarative project fails to compile when the OpenAPI contract interface is annotated with `@Validation.Validated`, but the interface has no no-arg validated property methods. The generated contract has validation constraints on operation parameters instead.


The generated project inside the reproducer is:

`target/generated-sources/openapi`

The relevant generated source is:

```java
@SuppressWarnings("deprecation")
@Validation.Validated
@Http.Path("/store/items")
public interface DefaultApi {
    @Http.GET
    @Http.Produces(MediaTypes.TEXT_PLAIN_VALUE)
    String listStoreItems(@Validation.Integer.Min(1)
                          @Validation.Integer.Max(100)
                          @Http.QueryParam("pageSize") Optional<Integer> pageSize,
                          @Http.QueryParam("pageToken") Optional<String> pageToken);
}
```

`ValidatedTypeGenerator` then generates a type validator for `DefaultApi` even though the interface has no validated no-arg property methods. The generated property-check methods contain a switch with only a throwing default branch, followed by `};`.

Generated output:

```java
@Override
public void check(ValidationContext ctx, DefaultApi instance, String propertyName) {
    switch (propertyName) {
        default -> throw new ValidationException("Invalid property name: " + propertyName);
        };
}

@Override
public void checkProperty(ValidationContext ctx, String propertyName, Object value) {
    switch (propertyName) {
        default -> throw new ValidationException("Invalid property name: " + propertyName);
        };
}
```

Javac reports the generated `};` empty statement as unreachable.

## Validation

Validated on current `helidon-4.x` checkout:

- Branch: `helidon-4.x`
- HEAD: `872ef7cb44`
- Generated project POM changed only from `4.4.1` to `4.5.0-SNAPSHOT` in the temporary extracted copy, so it uses the current branch artifacts.

Failure:

```text
[ERROR] /tmp/helidon-openapi-generated-project.AUuA70/target/generated-sources/openapi/target/generated-sources/annotations/com/example/api/DefaultApi__Validated.java:[27,14] unreachable statement
[ERROR] /tmp/helidon-openapi-generated-project.AUuA70/target/generated-sources/openapi/target/generated-sources/annotations/com/example/api/DefaultApi__Validated.java:[34,14] unreachable statement
```

## Suspected Cause

`declarative/codegen/src/main/java/io/helidon/declarative/codegen/validation/ValidatedTypeGenerator.java` always emits the `check(..., propertyName)` and `checkProperty(...)` switch forms with `addContentLine("};")`, even when the discovered property list is empty.

For a `@Validation.Validated` interface whose only validation annotations are on method parameters, `processValidatedInterface(...)` leaves `properties` empty, but `addCheckWithPropertyNameMethods(...)` still emits the invalid switch form.

## Expected

Generated validation code should compile for a declarative OpenAPI interface whose constraints are on operation parameters.

Possible fixes:

- Emit a normal switch statement terminator that does not introduce an unreachable empty statement.
- Or handle the empty-property case explicitly.
- Or avoid generating a type validator for interfaces with no type/property validation work.

--- a/declarative/codegen/src/main/java/io/helidon/declarative/codegen/validation/ValidatedTypeGenerator.java
+++ b/declarative/codegen/src/main/java/io/helidon/declarative/codegen/validation/ValidatedTypeGenerator.java
@@ -395,7 +395,7 @@ class ValidatedTypeGenerator {
                 .addContent("(")
                 .addContentLiteral("Invalid property name: ")
                 .addContentLine(" + propertyName);")
-                .addContentLine("};");
+                .addContentLine("}");
 
         classModel.addMethod(checkProperty);
 
@@ -428,7 +428,7 @@ class ValidatedTypeGenerator {
                 .addContent("(")
                 .addContentLiteral("Invalid property name: ")
                 .addContentLine(" + propertyName);")
-                .addContentLine("};");
+                .addContentLine("}");
 
         classModel.addMethod(checkPropertyValue);
     }

--- a/declarative/tests/codegen/src/test/java/io/helidon/declarative/codegen/validation/ValidationCodegenTest.java
+++ b/declarative/tests/codegen/src/test/java/io/helidon/declarative/codegen/validation/ValidationCodegenTest.java
@@ -84,6 +84,44 @@ public class ValidationCodegenTest {
     }
 
     @Test
+    void testValidatedInterfaceWithOnlyParameterConstraints() throws IOException {
+        var result = TestCompiler.builder()
+                .currentRelease()
+                .addClasspath(CLASSPATH)
+                .addProcessor(AptProcessor::new)
+                .addSource("DefaultApi.java", """
+                        package com.example;
+
+                        import java.util.Optional;
+
+                        import io.helidon.validation.Validation;
+
+                        @Validation.Validated
+                        public interface DefaultApi {
+                            String listStoreItems(@Validation.Integer.Min(1)
+                                                  @Validation.Integer.Max(100) Optional<Integer> pageSize);
+                        }
+                        """)
+                .build()
+                .compile();
+
+        String diagnostics = String.join("\n", result.diagnostics());
+        assertThat(diagnostics, result.success(), is(true));
+        assertThat(diagnostics, not(containsString("unreachable statement")));
+
+        var validator = result.sourceOutput().resolve("com/example/DefaultApi__Validated.java");
+        assertThat(Files.exists(validator), is(true));
+
+        var content = Files.readString(validator, StandardCharsets.UTF_8);
+        assertThat(content, matches("""
+                                            //...
+                                            class DefaultApi__Validated implements TypeValidator<DefaultApi> {
+                                            //...
+                                            }
+                                            """));
+    }
+
+    @Test
     void testPrivateFieldValidation() throws IOException {
         var result = TestCompiler.builder()
                 .currentRelease()

